### ISSUE TITLE
AutocompleteInput does not support dictionaries for options

### ISSUE THREAD

**Issue by cdanis** [holoviz/panel#6914]
The other single-option-select widgets (Select, DiscreteSlider, etc) support an options parameter that can be either a list or a dict.  AutocompleteInput's documentation claims that it has a compatible API with these widgets

However AutocompleteInput defines its options param as only a list.

```ValueError: List parameter 'AutocompleteInput.options' must be a list, not an object of <class 'dict'>.```

**Comment by ahuang11** [comment]
Thanks for the report!

I think this is a docs issue and a Bokeh upstream issue.

[AutoCompleteInput uses Bokeh's](https://github.com/holoviz/panel/blob/42d6cc0ba3d4d40261144ea7037bf6909dddbbf7/panel/widgets/select.py#L949) implementation underneath, and `completions` only support Lists. https://docs.bokeh.org/en/latest/docs/reference/models/widgets/inputs.html#bokeh.models.AutocompleteInput.completions

I think you can probably do your own conversion from key to value (or vice versa) inside a pn.bind?

**Comment by philippjfr** [comment]
>AutoCompleteInput uses Bokeh's implementation underneath, and completions only support Lists. https://docs.bokeh.org/en/latest/docs/reference/models/widgets/inputs.html#bokeh.models.AutocompleteInput.completions

This is true for other widgets as well but we handle the dictionary support on the Python side, we should do the same here.
